cvat-core/src/server-proxy.js

```diff
@@ -499,6 +499,47 @@
                 };
             }
 
+            async function importDataset(id, format, file, callback) {
+                const { backendAPI } = config;
+                const url = `${backendAPI}/projects/${id}/dataset`;
+
+                const formData = new FormData();
+                formData.append('dataset_file', file);
+
+                return new Promise((resolve, reject) => {
+                    async function request() {
+                        try {
+                            const response = await Axios.get(`${url}?action=import_status`, {
+                                proxy: config.proxy,
+                            });
+                            if (response.status === 202) {
+                                if (typeof callback === 'function') callback(response.data);
+                                setTimeout(request, 3000);
+                            } else if (response.status === 201) {
+                                resolve();
+                            } else {
+                                reject(generateError(response));
+                            }
+                        } catch (error) {
+                            if (error.response.status === 404) {
+                                try {
+                                    await Axios.post(`${url}?format=${format}`, formData, {
+                                        proxy: config.proxy,
+                                    });
+                                    setTimeout(request, 2000);
+                                } catch (_error) {
+                                    reject(generateError(error));
+                                }
```

Comment:
I would suggest to put it before getting status for the first time. In this way we always do unnesassary request first.